pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -592,20 +614,44 @@ func (p *Provisioner) setAdditionalSettings(options *apibkt.BucketOptions) error
 
 		maxSizeInt64, err = toInt64(maxSize)
 		if err != nil {
-			return errors.Wrapf(err, "failed to parse maxSize quota for user %q", p.cephUserName)
+			return errors.Wrapf(err, "failed to parse maxSize quota for user %q or bucket %q", p.cephUserName, p.bucketName)
 		}
 	}
 
-	objectUser, err := p.adminOpsClient.GetUser(p.clusterInfo.Context, admin.User{ID: p.cephUserName})
-	if err != nil {
-		return errors.Wrapf(err, "failed to fetch user %q", p.cephUserName)
-	}
-
-	// enable or disable quota for user
-	if *objectUser.UserQuota.Enabled != quotaEnabled {
-		err = p.adminOpsClient.SetUserQuota(p.clusterInfo.Context, admin.QuotaSpec{UID: p.cephUserName, Enabled: &quotaEnabled})
+	var (
+		bucket     admin.Bucket
+		objectUser admin.User
+	)
+	if UserID(options.ObjectBucketClaim.Spec.AdditionalConfig) != "" {
```

Comment:
Why quota features is different for this case, this bucket can be shared via another OBC right?

Reply:
Sorry, I don't get it. The difference here is that if the user specifies a userID, the bucket quota is set. If the user does not specify a userID, it remains the same as before, and the user quota is set.

Reply:
The quota under the "options.ObjectBucketClaim.Spec.AdditionalConfig" should be used for bucket, not for user ; why we need to set user'quoat with the "options.ObjectBucketClaim.Spec.AdditionalConfig" ? @thotz

Reply:
@microyahoo @zhucan The main reason behind this PR, the relationship between OBC and the ceph object store user was 1:1 but with buckets can be 1:many.  And if set quota on OBC then it should impact applications using that OBC, not all the applications using that bucket(like I said bucket can be shared among different OBCs). 
The second reason AFAIK the bucket quota in rgw effect only on owner of the bucket, so for brownfield cases(existing buckets) Rook only creates a new user and adds the policy to access the bucket for that user.  Then bucket quota won't effective against brownfield cases

Reply:
obc and ob is not 1:1?  if the ob is exists, when the obc bound to the exists ob, if the quota of the bucket is not equal to the obc, we should use the quota of the obc to update the quota of the ob. @thotz

Reply:
OB and OBC are k8s objects they are like PV and PVC. I am talking about backend RGW bucket the relation is not 1:1. OBC is not bucket IMO, more like access to the bucket. Thats in COSI we redefined Bucket and BucketAccess as separate definition rather that tied up with single object